services/frontend/src/app/RootApi.ts

```diff
@@ -1,10 +1,19 @@
 // Or from '@reduxjs/toolkit/query' if not using the auto-generated hooks
 import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 
-// initialize an empty api service that we'll inject endpoints into later as needed
+import type { RootState } from "./store";
+
 export const rootApi = createApi({
   baseQuery: fetchBaseQuery({
     baseUrl: "http://",
+    prepareHeaders: (headers, { getState }) => {
+      const state = getState() as RootState;
+      const token = state.internal.auth.sessionToken;
+      if (token) {
```

Comment:
Before this, should we check if the token is expired before adding it to the headers?

Reply:
I think this is not needed as the authentication layer will do it